magicctapipe/scripts/lst1_magic/lst1_magic_create_irf.py

```diff
@@ -97,50 +107,99 @@ def create_irf(
     )
 
     # Load the input gamma MC DL2 data file
     logger.info(f"\nInput gamma MC DL2 data file:\n{input_file_gamma}")
 
     event_table_gamma, pnt_gamma, sim_info_gamma = load_mc_dl2_data_file(
-        input_file_gamma, quality_cuts, irf_type, dl2_weight_type
+        input_file_gamma, quality_cuts, event_type, dl2_weight_type
     )
 
-    mean_offset = event_table_gamma["true_source_fov_offset"].to(u.deg).mean().round(1)
-    fov_offset_bins = u.Quantity([mean_offset - 0.1 * u.deg, mean_offset + 0.1 * u.deg])
+    is_diffuse_mc = sim_info_gamma.viewcone.to_value(u.deg) > 0
+    logger.info(f"\nIs diffuse MC: {is_diffuse_mc}")
 
-    logger.info(
-        f"\nMean FoV offset: {mean_offset}" f"\n--> FoV offset bins: {fov_offset_bins}"
-    )
+    if is_diffuse_mc:
 
-    # Load the background data files
-    event_table_bkg = QTable()
+        logger.info("\nCreating FoV offset bins from the configuration...")
 
-    is_proton_mc = input_file_proton is not None
-    is_electron_mc = input_file_electron is not None
+        config_fov_bins = config_irf["fov_offset_bins"]
+
+        fov_bins_start = u.Quantity(config_fov_bins["start"])
+        fov_bins_stop = u.Quantity(config_fov_bins["stop"])
+        fov_bins_n_edges = config_fov_bins["n_edges"]
 
-    if is_proton_mc and (not is_electron_mc):
         logger.info(
-            "\nWARNING: Proton MC exists but electron MC does not. "
-            "Skipping the creation of the background model..."
+            "\nFov offset bins (linear scale):"
+            f"start: {fov_bins_start}"
+            f"stop: {fov_bins_stop}"
+            f"n_edges: {fov_bins_n_edges}"
+        )
+
+        fov_offset_bins = u.Quantity(
+            value=np.linspace(
+                start=fov_bins_start.to_value(u.deg).round(1),
+                stop=fov_bins_stop.to_value(u.deg).round(1),
+                num=fov_bins_n_edges,
+            ),
+            unit=u.deg,
         )
 
-    elif is_electron_mc and (not is_proton_mc):
+    else:
+        logger.info("\nCreating FoV offset bins from the true FoV offset...")
+
+        true_fov_offset = event_table_gamma["true_source_fov_offset"].to(u.deg)
+        mean_true_fov_offset = true_fov_offset.mean().round(1)
+
+        fov_offset_bins = u.Quantity([-0.1, 0.1], u.deg) + mean_true_fov_offset
+
         logger.info(
-            "\nWARNING: Electron MC exists but proton MC does not. "
-            "Skipping the creation of the background model..."
+            f"\nMean true FoV offset: {mean_true_fov_offset}"
+            f"\n--> FoV offset bins: {fov_offset_bins}"
+        )
+
+    # Here we decide the IRF type based on the number of FoV offset
+    # bins, "POINT-LIKE" in case of 1 bin and "FULL-ENCLOSURE" in the
+    # other cases. It allows for creating the "POINT-LIKE" IRFs with
+    # diffuse gamma MCs by selecting a given FoV offset region.
+
+    is_point_like = len(fov_offset_bins) == 2  # 1 bin -> n_edges = 2
```

Comment:
here the description and the following line do not match. maybe you can define:

```
is_point_like = (len(fov_offset_bins) - 1) == 1
```

Reply:
Thanks, a bit differently but I changed it.